metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -208,15 +211,90 @@ func (sink *StackdriverSink) ExportData(dataBatch *core.DataBatch) {
 				req.TimeSeries = append(req.TimeSeries, point)
 			}
 			if len(req.TimeSeries) >= maxTimeseriesPerRequest {
-				sink.requestQueue <- req
+				requests = append(requests, req)
 				req = getReq()
 			}
 		}
 	}
 
 	if len(req.TimeSeries) > 0 {
-		sink.requestQueue <- req
+		requests = append(requests, req)
 	}
+
+	go sink.sendRequests(requests)
+}
+
+func (sink *StackdriverSink) sendRequests(requests []*sd_api.CreateTimeSeriesRequest) {
+	// Each worker can handle at least batchExportTimeout/sdRequestLatencySec requests within the specified period.
+	// 5 extra workers just in case.
+	workers := 5 + len(requests)/(sink.batchExportTimeoutSec/sdRequestLatencySec)
```

Comment:
Because the whole metric pass assumes that it is synchronous. We don't want to start another metric scraping if there are too many pending sink sync operations. Heapster, especially in large  clusters, has memory limits set with assumption that there are at most 2 (or 3, don't remember exactly) full scrape results in the memory. Having more may cross the limit and if unlucky with free resources, crash the whole service.